### PR TITLE
Gangwei/Change address to type

### LINKED ISSUE #120: Change Address field to Type

### PR DESCRIPTION
**Pull request by nuovonatura**
Resolve #120

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -62,11 +62,11 @@ The sections below give more details of each component.
...
 
 The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
 
-The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
+The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/type/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
```
Comment (dearvae) [inline docs/DeveloperGuide.md]: i don't think you suppose to change this address here

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -9,7 +9,7 @@
     public static final Prefix PREFIX_TITLE = new Prefix("title:");
     public static final Prefix PREFIX_PHONE = new Prefix("p/");
     public static final Prefix PREFIX_DESCRIPTION = new Prefix("desc:");
-    public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
+    public static final Prefix PREFIX_TYPE = new Prefix("a/");
```
Comment (GabriellaTeh) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: you forgot to change the prefix, i think it should be type:

Comment (dearvae) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: line 12, the prefix for type is "type:"

File: src/test/java/seedu/address/model/TypeBookTest.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 import seedu.address.model.task.exceptions.DuplicateTaskException;
 import seedu.address.testutil.TaskBuilder;
 
-public class AddressBookTest {
+public class TypeBookTest {
```
Comment (GabriellaTeh) [inline src/test/java/seedu/address/model/TypeBookTest.java]: I dont think addressbook should be changed to typebook

File: src/test/java/seedu/address/storage/JsonSerializableTypeBookTest.java [hunk 4]
```diff
@@ -13,7 +13,7 @@
 import seedu.address.model.AddressBook;
 import seedu.address.testutil.TypicalTasks;
 
-public class JsonSerializableAddressBookTest {
+public class JsonSerializableTypeBookTest {
```
Comment (GabriellaTeh) [inline src/test/java/seedu/address/storage/JsonSerializableTypeBookTest.java]: This one also, should not change addressbook to typebook

File: src/test/java/seedu/address/storage/JsonTypeBookStorageTest.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 import seedu.address.model.AddressBook;
 import seedu.address.model.ReadOnlyAddressBook;
 
-public class JsonAddressBookStorageTest {
+public class JsonTypeBookStorageTest {
```
Comment (GabriellaTeh) [inline src/test/java/seedu/address/storage/JsonTypeBookStorageTest.java]: this one also, should not change

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 6]
```diff
@@ -17,5 +17,5 @@ Person -up-> "*" Tag
 Person *--> Name
 Person *--> Phone
 Person *--> Email
-Person *--> Address
+Person *--> Type
```
Comment (BobbyZhouZijian) [inline docs/diagrams/BetterModelClassDiagram.puml:20]: Nice that you changed the UML as well.

File: src/main/java/seedu/address/model/task/Description.java [hunk 7]
```diff
@@ -16,7 +16,7 @@
             + "example => desc:Tell Zijian to review my PR!!! Repeat, this is urgent!!! You get it?";
 
     /*
-     * The first character of the address must not be a whitespace,
+     * The first character of the type must not be a whitespace,
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Description.java]: oops, I think this should have been "description". My bad I forgot to change this. Can you change it?

File: src/main/java/seedu/address/model/task/Type.java [hunk 8]
```diff
@@ -4,36 +4,36 @@
...
- * Represents a Task's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Task's type in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidType(String)}
  */
-public class Address {
+public class Type {
 
-    public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Type can take any strings, and it should not be blank";
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/task/Type.java]: Should we really allow type to be any string? Or just one word letters connected by '-'

**Review by BobbyZhouZijian — COMMENTED** [review]
Some small things to fix. Others LGTM. For the type, was wondering if we really should allow all characters?

**Review by BobbyZhouZijian — APPROVED** [review]
LGTM

**Pull request merged**